docs/documenting/yaml-style-guide.md

```diff
@@ -0,0 +1,483 @@
+available and is the one that should be used.
+
+```yaml
+# Good
+action:
+  - service: light.turn_on
+    target:
+      entity_id: light.living_room
+  - service: light.turn_on
+    target:
+      area_id: light.living_room
+  - service: light.turn_on
+    target:
+      area_id: living_room
+      entity_id: light.office_desk
+      device_id: 21349287492398472398
+
+# Bad
+action:
+  - service: light.turn_on
+    entity_id: light.living_room
+  - service: light.turn_on
+     data:
+       entity_id: light.living_room
+```
+
+### Properties that accept a scalar or a list of scalars
+
+Home Assistant has a lot of places that access both a scalar value or a list
+of scalar values. Additionally, sometimes, it even accepts a comma-separated
+string value as a list.
+
+The following applies in cases a single value or a list of scalar values
+is accepted:
+
+- Putting multiple values in a single scalar value (comma separated string)
+  must not be used.
+- If a list is used, it must be block style.
+- A list with a single scalar value should not be used.
+- The use of a single scalar value is allowed.
+
+```yaml
+# Good
+entity_id: light.living_room
+entity_id:
+  - light.living_room
+  - light.office
+
+# Bad
+entity_id: light.living_room, light.office
+entity_id: [light.living_room, light.office]
+entity_id:
+  - light.living_room
+```
+
+### Properties that accept a mapping or a list of mappings
+
+Home Assistant has properties that accept both a mapping or a list of mappings.
+Well known examples are: `condition`, `action`, `sequence`.
+
+In case a property accepts a single mapping or a list of mappings, a list of
+mapping must be used. Even when a single mapping is passed in.
+
+This makes it easier to understand one can add more items to it and easier to
+copy and paste a single item into your own code.
+
+```yaml
+# Good
+action:
+  - service: light.turn_on
+    target:
+      entity_id: light.living_room
+
+# Bad
+action:
+  service: light.turn_on
+  target:
+    entity_id: light.living_room
+```
+
+### Templates
+
+Home Assistant templates are powerful, but they can be really confusing or hard
+to understand for a less experienced user. Therefore, the use of templates
+should be avoided if a pure YAML version available.
+
+Additionally, the use of template requires additional escaping in our
+documentation to avoid our website code to confuse it for the Liquid syntax.
+Avoiding templates in general removes the need of additional escaping.
+
+```yaml
+# Good
+condition:
+  - condition: numeric_state
+    entity_id: sun.sun
+    attribute: elevation
+    below: 4
+
+# Bad
+condition:
+  - condition: template
+    value_template: "{{ state_attr('sun.sun', 'elevation') < 4 }}"
+```
+
+#### Quoting style
+
+Templates are strings, and thus are double-quoted. As a result of that,
+single quotes should be used inside the template.
+
+```yaml
+# Good
+condition:
+  - "{{ 'some_value' == some_other_value }}" 
```

Comment:
Should this not be `condition: "{{ 'some_value' == some_other_value }}"` per the above on scalar/list cases?